### PR TITLE
Port ErrorPagination

### PR DESCRIPTION
**Pull request by devjiwonchoi**
Ported `ErrorPagination` and replaced in the `Errors` component.

### CODE REVIEW

**Comment by devjiwonchoi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/vercel/next.js/74097?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#74098** <a href="https://app.graphite.dev/github/pr/vercel/next.js/74098?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#74097** <a href="https://app.graphite.dev/github/pr/vercel/next.js/74097?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/vercel/next.js/74097?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#74090** <a href="https://app.graphite.dev/github/pr/vercel/next.js/74090?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#74089** <a href="https://app.graphite.dev/github/pr/vercel/next.js/74089?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#74073** <a href="https://app.graphite.dev/github/pr/vercel/next.js/74073?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| buildDuration | 19.1s | 17.1s | N/A |
| buildDurationCached | 16.1s | 13.6s | N/A |
| nodeModulesSize | 416 MB | 416 MB | ⚠️ +139 kB |
| nextStartRea..uration (ms) | 490ms | 488ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| 1187-HASH.js gzip | 52.4 kB | 52.4 kB | N/A |
| 8276.HASH.js gzip | 169 B | 168 B | N/A |
| 8377-HASH.js gzip | 5.36 kB | 5.36 kB | N/A |
| bccd1874-HASH.js gzip | 53 kB | 53 kB | N/A |
| framework-HASH.js gzip | 57.5 kB | 57.5 kB | N/A |
| main-app-HASH.js gzip | 232 B | 235 B | N/A |
| main-HASH.js gzip | 34.1 kB | 34.1 kB | N/A |
| webpack-HASH.js gzip | 1.71 kB | 1.71 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 39.4 kB | 39.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| _app-HASH.js gzip | 193 B | 193 B | ✓ |
| _error-HASH.js gzip | 193 B | 193 B | ✓ |
| amp-HASH.js gzip | 512 B | 510 B | N/A |
| css-HASH.js gzip | 343 B | 342 B | N/A |
| dynamic-HASH.js gzip | 1.84 kB | 1.84 kB | ✓ |
| edge-ssr-HASH.js gzip | 265 B | 265 B | ✓ |
| head-HASH.js gzip | 363 B | 362 B | N/A |
| hooks-HASH.js gzip | 393 B | 392 B | N/A |
| image-HASH.js gzip | 4.49 kB | 4.49 kB | N/A |
| index-HASH.js gzip | 268 B | 268 B | ✓ |
| link-HASH.js gzip | 2.35 kB | 2.34 kB | N/A |
| routerDirect..HASH.js gzip | 328 B | 328 B | ✓ |
| script-HASH.js gzip | 397 B | 397 B | ✓ |
| withRouter-HASH.js gzip | 323 B | 326 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.59 kB | 3.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| _buildManifest.js gzip | 749 B | 746 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| index.html gzip | 523 B | 524 B | N/A |
| link.html gzip | 538 B | 538 B | ✓ |
| withRouter.html gzip | 519 B | 521 B | N/A |
| Overall change | 538 B | 538 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| edge-ssr.js gzip | 129 kB | 129 kB | N/A |
| page.js gzip | 205 kB | 205 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 667 B | 668 B | N/A |
| middleware-r..fest.js gzip | 155 B | 156 B | N/A |
| middleware.js gzip | 31.3 kB | 31.3 kB | N/A |
| edge-runtime..pack.js gzip | 844 B | 844 B | ✓ |
| Overall change | 844 B | 844 B | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| 274-experime...dev.js gzip | 322 B | 322 B | ✓ |
| 274.runtime.dev.js gzip | 314 B | 314 B | ✓ |
| app-page-exp...dev.js gzip | 356 kB | 356 kB | ⚠️ +469 B |
| app-page-exp..prod.js gzip | 129 kB | 129 kB | ✓ |
| app-page-tur..prod.js gzip | 142 kB | 142 kB | ✓ |
| app-page-tur..prod.js gzip | 137 kB | 137 kB | ✓ |
| app-page.run...dev.js gzip | 346 kB | 347 kB | ⚠️ +489 B |
| app-page.run..prod.js gzip | 125 kB | 125 kB | ✓ |
| app-route-ex...dev.js gzip | 37.5 kB | 37.5 kB | ✓ |
| app-route-ex..prod.js gzip | 25.5 kB | 25.5 kB | ✓ |
| app-route-tu..prod.js gzip | 25.5 kB | 25.5 kB | ✓ |
| app-route-tu..prod.js gzip | 25.3 kB | 25.3 kB | ✓ |
| app-route.ru...dev.js gzip | 39.1 kB | 39.1 kB | ✓ |
| app-route.ru..prod.js gzip | 25.3 kB | 25.3 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.69 kB | 9.69 kB | ✓ |
| pages-api.ru...dev.js gzip | 11.6 kB | 11.6 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.68 kB | 9.68 kB | ✓ |
| pages-turbo...prod.js gzip | 21.7 kB | 21.7 kB | ✓ |
| pages.runtim...dev.js gzip | 27.5 kB | 27.5 kB | ✓ |
| pages.runtim..prod.js gzip | 21.7 kB | 21.7 kB | ✓ |
| server.runti..prod.js gzip | 916 kB | 916 kB | ✓ |
| Overall change | 2.43 MB | 2.43 MB | ⚠️ +958 B |

</details>

<details>
<summary><strong>build cache</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 12-19-port_errorpagination | Change |
| - | - | - | - |
| 0.pack gzip | 2.08 MB | 2.08 MB | N/A |
| index.pack gzip | 74.2 kB | 74.2 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

```diff
failed to diff
```
</details>

</details>

</details>
Commit: eec18eed8f6fc19e064552d0a5fa05fe26262a65

**Review by gaojude — APPROVED** [review]

**Comment by devjiwonchoi** [comment]
### Merge activity

* **Dec 19, 3:48 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/vercel/next.js/74097).

**Comment by ijjk** [comment]
## Failing test suites 
Commit: a9a4d6b76d3f8f44fcbb09c712b0f6c56904b26a

`__NEXT_EXPERIMENTAL_PPR=true pnpm test-dev test/e2e/app-dir/non-root-project-monorepo/non-root-project-monorepo.test.ts` (PPR)
- non-root-project-monorepo > source-maps > should work on RSC
- non-root-project-monorepo > source-maps > should work on SSR
- non-root-project-monorepo > source-maps > should work on client-side

<details>
<summary>Expand output</summary>

  ● non-root-project-monorepo › source-maps › should work on RSC

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `non-root-project-monorepo source-maps should work on RSC 1`

    - Snapshot  - 1
    + Received  + 1

    - app/separate-file.ts (1:11) @ Error
    + app/separate-file.ts (1:11) @ eval

      > 1 | throw new Error('Expected error')
          |           ^
        2 |

      89 |         } else {
      90 |           // TODO the function name is incorrect
    > 91 |           expect(await getRedboxSource(browser)).toMatchInlineSnapshot(`
         |                                                  ^
      92 |             "app/separate-file.ts (1:11) @ Error
      93 |
      94 |             > 1 | throw new Error('Expected error')

      at Object.toMatchInlineSnapshot (e2e/app-dir/non-root-project-monorepo/non-root-project-monorepo.test.ts:91:50)

  ● non-root-project-monorepo › source-maps › should work on RSC

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `non-root-project-monorepo source-maps should work on RSC 2`

    - Snapshot  - 3
    + Received  + 3

      <unknown>
      rsc)/./app/separate-file.ts (rsc://React/Server/file://<full-path>/apps/web/.next/server/app/source-maps-rsc/page.js
      __webpack_require__
      file://<full-path>/apps/web/.next/server/webpack-runtime.js
    - require
    + innerArrowFunction
      app/source-maps-rsc/page.tsx (14:3)
    - innerArrowFunction
    + innerFunction
      app/source-maps-rsc/page.tsx (10:3)
    - innerFunction
    + Page
      app/source-maps-rsc/page.tsx (4:5)

       98 |           // TODO webpack runtime code shouldn't be included in stack trace
       99 |           expect(normalizeStackTrace(await getRedboxCallStack(browser)))
    > 100 |             .toMatchInlineSnapshot(`
          |              ^
      101 |            "<unknown>
      102 |            rsc)/./app/separate-file.ts (rsc://React/Server/file://<full-path>/apps/web/.next/server/app/source-maps-rsc/page.js
      103 |            __webpack_require__

      at Object.toMatchInlineSnapshot (e2e/app-dir/non-root-project-monorepo/non-root-project-monorepo.test.ts:100:14)

  ● non-root-project-monorepo › source-maps › should work on SSR

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `non-root-project-monorepo source-maps should work on SSR 2`

    - Snapshot  - 3
    + Received  + 3

    @@ -4,11 +4,11 @@
      file://<full-path>/apps/web/.next/static/chunks/webpack.js (700:31)
      __webpack_require__
      file://<full-path>/apps/web/.next/static/chunks/webpack.js (37:33)
      fn
      file://<full-path>/apps/web/.next/static/chunks/webpack.js (357:21)
    - require
    + innerArrowFunction
      app/source-maps-ssr/page.tsx (16:3)
    - innerArrowFunction
    + innerFunction
      app/source-maps-ssr/page.tsx (12:3)
    - innerFunction
    + Page
      app/source-maps-ssr/page.tsx (6:5)

      147 |           // TODO webpack runtime code shouldn't be included in stack trace
      148 |           expect(normalizeStackTrace(await getRedboxCallStack(browser)))
    > 149 |             .toMatchInlineSnapshot(`
          |              ^
      150 |            "./app/separate-file.ts
      151 |            file://<full-path>/apps/web/.next/static/chunks/app/source-maps-ssr/page.js (27:1)
      152 |            options.factory

      at Object.toMatchInlineSnapshot (e2e/app-dir/non-root-project-monorepo/non-root-project-monorepo.test.ts:149:14)

  ● non-root-project-monorepo › source-maps › should work on client-side

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `non-root-project-monorepo source-maps should work on client-side 2`

    - Snapshot  - 3
    + Received  + 3

    @@ -4,11 +4,11 @@
      file://<full-path>/apps/web/.next/static/chunks/webpack.js (712:31)
      __webpack_require__
      file://<full-path>/apps/web/.next/static/chunks/webpack.js (37:33)
      fn
      file://<full-path>/apps/web/.next/static/chunks/webpack.js (369:21)
    - require
    + innerArrowFunction
      app/source-maps-client/page.tsx (17:3)
    - innerArrowFunction
    + innerFunction
      app/source-maps-client/page.tsx (13:3)
    - innerFunction
    + effectCallback
      app/source-maps-client/page.tsx (7:5)

      200 |           // TODO webpack runtime code shouldn't be included in stack trace
      201 |           expect(normalizeStackTrace(await getRedboxCallStack(browser)))
    > 202 |             .toMatchInlineSnapshot(`
          |              ^
      203 |             "./app/separate-file.ts
      204 |             file://<full-path>/apps/web/.next/static/chunks/app/source-maps-client/page.js (27:1)
      205 |             options.factory

      at Object.toMatchInlineSnapshot (e2e/app-dir/non-root-project-monorepo/non-root-project-monorepo.test.ts:202:14)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/client/components/react-dev-overlay/_experimental/internal/components/LeftRightDialogHeader/LeftRightDialogHeader.tsx [hunk 1]
```diff
@@ -1,169 +0,0 @@
...
-            data-nextjs-errors-dialog-left-right-close-button
-            ref={buttonClose}
-            type="button"
-            onClick={close}
-            aria-label="Close"
-          >
-            <span aria-hidden="true">
-              <CloseIcon />
-            </span>
-          </button>
```
Comment (devjiwonchoi) [inline packages/next/src/client/components/react-dev-overlay/_experimental/internal/components/LeftRightDialogHeader/LeftRightDialogHeader.tsx:163]: @gaojude @huozhi 
FYI omitted this Close button as we don't have it in our design plan. Would it be better to do it in a separate stack?

**Pull request merged**